pandas/core/arrays/base.py

```diff
@@ -354,6 +355,19 @@ def __iter__(self):
         for i in range(len(self)):
             yield self[i]
 
+    def __contains__(self, item) -> bool:
+        """
+        Return for `item in self`.
+        """
+        # comparisons of any item to pd.NA always return pd.NA, so e.g. "a" in [pd.NA]
+        # would raise a TypeError. The implementation below works around that.
+        if item is self.dtype.na_value:
+            return isna(self).any() if self._can_hold_na else False
+        elif is_scalar(item) and isna(item):
+            return False
```

Comment:
This could maybe be done in a separate PR, as it is FloatingArray specific (and the NaN behaviour is not yet fully fleshed out), but this check will do the wrong thing for `np.nan` if the FloatingArray actually contains a NaN (as `self.dtype._na_value` is then pd.NA and not np.nan).

Reply:
So a `FloatingArray` can have multiple nan types? Is that really necessary?

I think I would prefer that would be handled in `FloatingArray`, so I don't have to stretch this PR too far.

Reply:
> So a FloatingArray can have multiple nan types? Is that really necessary?

Not multiple "nan types", to be nitpicky, since pd.NA is not a "nan". But yes, it can contains both pd.NA and np.nan at the moment. But that's not yet fully decided though, long discussion at https://github.com/pandas-dev/pandas/issues/32265/. Happy to hear your thoughts about it.

> I think I would prefer that would be handled in FloatingArray, so I don't have to stretch this PR too far.

Yes, will put it on my list of follow to-do items for FloatingArray that I need to open. So don't worry about it here